### PR TITLE
Added quick logging mechanism, to be changed later

### PR DESCRIPTION
**Pull request by TamerShlash**
Quick as possible, this mechanism creates a logger in the pipeline and assigns it to each step, so that all the logging will happen on the same logger.

### CODE REVIEW

File: lib/optimus_prime/step.rb [hunk 1]
```diff
@@ -35,8 +35,10 @@ def subclasses
       end
     end
 
-    def logger
-      @logger ||= Logger.new(STDERR)
+    attr_reader :logger
```
Comment (omarkhan) [inline lib/optimus_prime/step.rb]: Replace `attr_reader` with `attr_accessor` and you can drop the setter method

File: lib/optimus_prime/pipeline.rb [hunk 2]
```diff
@@ -35,6 +35,10 @@ def join
 
     def steps
       @steps ||= graph.map { |key, step| [key, Step.create(step)] }.to_h
+                      .each do |key, step|
+                        step.logger=@logger
```
Comment (omarkhan) [inline lib/optimus_prime/pipeline.rb]: `step.logger=@logger` -> `step.logger = @logger`

If we switch rubocop off I don't want to do his job myself :smile:

File: lib/optimus_prime/step.rb (hunk 1 above)
Comment (TamerShlash) [inline lib/optimus_prime/step.rb]: Good idea :+1:

File: lib/optimus_prime/pipeline.rb (hunk 2 above)
Comment (TamerShlash) [inline lib/optimus_prime/pipeline.rb]: Actually you're better than he's :smiley:

File: lib/optimus_prime/pipeline.rb [hunk 3]
```diff
@@ -35,6 +37,10 @@ def join
 
     def steps
       @steps ||= graph.map { |key, step| [key, Step.create(step)] }.to_h
+                      .each do |key, step|
```
Comment (T-Dnzt) [inline lib/optimus_prime/pipeline.rb]: Why not put this on one line like the `map` above?

Comment (omarkhan) [inline lib/optimus_prime/pipeline.rb]: Also, the `.to_h` part should be at the end of the chain

Comment (TamerShlash) [inline lib/optimus_prime/pipeline.rb]: Good point :+1:

**Pull request merged**